src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -323,10 +333,74 @@ void net_device_table_mgr::verify_ipoib_mode(struct ifaddrs* ifa)
 		vlog_printf(VLOG_WARNING,"This option in no longer needed in this version\n");
 		vlog_printf(VLOG_WARNING,"Please refer to Release Notes for more information\n");
 		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		return 0;
 	}
 	else {
 		ndtm_logdbg("verified interface '%s' is running with umcast disabled", ifa->ifa_name);
 	}
+	return 1;
+}
+
+bool net_device_table_mgr::check_ib_device_supported(const char* ifname) {
+	if (check_device_exist(ifname, CX4_DRIVER_PATH)) {
+		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		vlog_printf(VLOG_WARNING,"VMA doesn't support IB on CX4\n");
+		vlog_printf(VLOG_WARNING,"All traffic over interface %s will not be offloaded!\n", ifname);
+		vlog_printf(VLOG_WARNING,"************************************************************************\n");
+		return 0;
+	} else {
+		return check_qp_creation(ifname, IBV_QPT_UD);
+	}
+}
+
+bool net_device_table_mgr::check_qp_creation(const char* ifname, enum ibv_qp_type qp_type) {
```

Comment:
return false instead of 0
return true instead of 1
This remark applies to all boolean functions